pegasus/helpers/section_api_helpers.rb

```diff
@@ -21,23 +21,29 @@ def self.fetch_user_students(user_id)
   def self.create(params)
     name = !params[:name].to_s.empty? ? params[:name].to_s : 'New Student'
     gender = valid_gender?(params[:gender]) ? params[:gender] : nil
+    provider = valid_provider?(params[:provider]) ? params[:provider] : 'sponsored'
     birthday = age_to_birthday(params[:age]) ?
       age_to_birthday(params[:age]) : params[:birthday]
 
     created_at = DateTime.now
 
-    row = Dashboard.db[:users].insert(
+    data =
       {
         name: name,
         user_type: 'student',
-        provider: 'sponsored',
+        provider: provider,
         gender: gender,
         birthday: birthday,
         created_at: created_at,
         updated_at: created_at,
         username: UserHelpers.generate_username(Dashboard.db[:users], name)
-      }.merge(random_secrets)
-    )
+      }
+    if provider == 'sponsored'
+      data.merge!(random_secrets)
```

Comment:
All users, whether or not the account is sponsored, should be assigned a `secret_picture_id` and `secret_words`.
```
SELECT secret_picture_id, COUNT(0) FROM users GROUP BY 1;
+-------------------+----------+
| secret_picture_id | COUNT(0) |
+-------------------+----------+
|              NULL |       23 |
|                 1 |  1124523 |
|                 2 |  1127598 |
|                 3 |  1141699 |
|                 4 |  1142563 |
|                 5 |  1131047 |
|                 6 |  1151284 |
|                 7 |  1133645 |
|                 8 |  1131632 |
|                 9 |  1122342 |
|                10 |  1126046 |
|                11 |  1139115 |
|                12 |  1126752 |
|                13 |  1151663 |
|                14 |  1124970 |
|                15 |  1113153 |
|                16 |  1141055 |
|                17 |  1134524 |
|                18 |  1140638 |
|                19 |  1123588 |
|                20 |  1109450 |
|                21 |  1125753 |
|                22 |  1117174 |
+-------------------+----------+
```
TODO(asher): Fix the twenty-three rows with NULLs.